src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -796,12 +812,14 @@ private struct PutCountersData
 
                         using (Slice.External(context.Allocator, kvp.Key, out var countersGroupKey))
                         {
-                            if (currentData.Size > MaxCounterDocumentSize)
+                            currentData.TryGet(Values, out BlittableJsonReaderObject localCounters);
+
+                            if (currentData.Size > MaxCounterDocumentSize &&
+                                localCounters.Count > 1)
```

Comment:
why are you checking the count here?

Reply:
if counter name is huge then it's possible to exceed MaxCounterDocumentSize while having just one counter in counter-group document

Reply:
I don't see how that can be. We should have a much lower limit to the size of a counter name